org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/ObjectComparision.java

```diff
@@ -0,0 +1,654 @@
+	public static final String SELECTED_OBJECT_2 = "Selection2"; //$NON-NLS-1$
+
+	public static final String KEYSET_1 = "keySet1"; //$NON-NLS-1$
+	public static final String KEYSET_2 = "keySet2"; //$NON-NLS-1$
+
+	public static final String VALUESET_1 = "valueSet1"; //$NON-NLS-1$
+	public static final String VALUESET_2 = "valueSet2"; //$NON-NLS-1$
+
+	/**
+	 * Extracts the actual value of the given object
+	 *
+	 * @param value
+	 *            IJavaValue of the selected object
+	 * @throws DebugException
+	 * @return returns actual value of the object or return combination object's name and id if extraction is not possible
+	 */
+	@SuppressWarnings("nls")
+	private String objectExtraction(IJavaValue value) throws DebugException {
+		if (value.getReferenceTypeName().equals("java.lang.String")) {
+			return value.toString();
+		}
+		if (value.getReferenceTypeName().startsWith("java.lang.StringBu")) {
+			return stringObjectExtraction(value);
+		}
+		for (IVariable v : value.getVariables()) {
+			if (v.getName().equalsIgnoreCase("value")) {
+				return v.getValue().getValueString();
+			}
+		}
+		return value.toString();
+	}
+
+	/**
+	 * Extracts string value of the given string object
+	 *
+	 * @param value
+	 *            IJavaValue of the selected object
+	 * @throws DebugException
+	 * @return returns actual value of the string or return combination object's name and id if extraction is not possible
+	 */
+	@SuppressWarnings("nls")
+	private String stringObjectExtraction(IJavaValue value) throws DebugException {
+		char c;
+		int i;
+		StringBuilder contentBuilder = new StringBuilder();
+		for (IVariable v : value.getVariables()) {
+			if (v.getName().equalsIgnoreCase("value")) {
+				for (IVariable inner : v.getValue().getVariables()) {
+					i = Integer.parseInt(inner.getValue().toString());
+					if (i == 0) {
+						break;
+					}
+					c = (char) i;
+					contentBuilder.append(c);
+				}
+			}
+		}
+		return contentBuilder.toString();
+	}
+
+	/**
+	 * Compare selected Set objects
+	 *
+	 * @param selectedObject1
+	 *            First selected object from variable view
+	 * @param selectedObject2
+	 *            Second selected object from variable view
+	 * @throws DebugException
+	 * @return returns a Map of comparison result details
+	 */
+	@SuppressWarnings("nls")
+	public Map<String, Object> compareSets(IJavaVariable selectedObject1, IJavaVariable selectedObject2) throws DebugException {
+		Map<String, Object> result = new HashMap<>();
+		result.put(SELECTED_OBJECT_1, selectedObject1.getName());
+		result.put(SELECTED_OBJECT_2, selectedObject2.getName());
+		if (selectedObject1.getValue() instanceof IJavaObject javaObject1 && selectedObject2.getValue() instanceof IJavaObject javaObject2) {
+			IJavaFieldVariable f1 = javaObject1.getField("map", false);
+			IJavaFieldVariable f2 = javaObject2.getField("map", false);
+			if (f1 == null & f2 == null && javaObject1.getReferenceTypeName().startsWith("java.util.TreeSet")) {
```

Comment:
Would be better to use  (more general) Interfaces like SortedSet instead of Implementations like TreeSet

Reply:
okay, will check for SortedSet